WIP: road to pair my TicWatch

deadYokai opened the pull request:
Just trying to make a pair with my watch, no more

Many parts of code is just stubs, or workarounds, needed proper implementation

~~After https://github.com/microg/GmsCore/pull/3204/commits/465f5f01967bcafa1982a2f1568bcbfbdf669013 requires merging PR https://github.com/microg/Wearable/pull/3  and pushing to maven~~ (not relevant after https://github.com/microg/GmsCore/pull/3204/commits/63a77a0ee8a0b8d349c63f6483f9c0204d031d9e)

P.S. maybe needed code cleanup

TODO:
- [ ] full pair process (partialy implemented)
- [ ] test watchfaces
- [ ] test sensors
- [ ] refactor/cleanup

deadYokai committed:
Added bare layout for wearable TOS

Added some callbacks

Added feature list to WearableService.java

Added android:exported="true", required by android 12, for TOS only, cuz i don't know if in other needed true or false

deadYokai committed:
Added founded callbacks to IWearableCallbacks.aidl

deadYokai committed:
Added missing classes (stubs) + aidls

deadYokai committed:
Added some aidl files and consent stuff

deadYokai commented:
now maybe fixes https://github.com/microg/GmsCore/issues/2444 ,

at least my Mobvoi Health going to pair screen (after TOS)

but code in rough state for now

deadYokai committed:
capability, connectionConfiguration rework

deadYokai committed:
added services `getNodeId`, `sendRequest` and option variants of `sendRequest`, `sendMessage`

prepairing ConnectionConfiguration to bluetooth stuff, to proper pair devices

filled some Parcable classes

deadYokai committed:
finaly successful connection
and handshake

some bluetooth stuff

deadYokai commented:
now need to implement openChannel method

deadYokai committed:
Some channel manager things

still failed to pair

deadYokai commented:
@ale5000-git as i mentioned before 

> After [465f5f0](https://github.com/microg/GmsCore/commit/465f5f01967bcafa1982a2f1568bcbfbdf669013) requires merging PR [microg/Wearable#3](https://github.com/microg/Wearable/pull/3) and pushing to maven

deadYokai commented:
or i can try to move  https://github.com/microg/Wearable/ into GmsCore repo and get rid of dependence 

https://mvnrepository.com/artifact/org.microg/wearable/0.1.1

ale5000-git commented:
Sorry I missed that point, then we have to wait for @mar-v-in for the decision of this thing.
In the meanwhile go ahead with the other parts you are working, thanks.

mar-v-in commented:
I'd suggest to entirely move the content of the microg/Wearable repo into play-services-wearable/core module in this repo and then we can archive the microg/Wearable repo entirely.

deadYokai committed:
Wearable lib relocation

and some bluetooth changes

deadYokai commented:
just sharing with good looking logs (logcat)

P.s. still not paired yet

<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/66aec7ef-fe64-4443-8777-94b03d93cb82" />

deadYokai committed:
updated NodeDatabaseHelper

some Bluetooth changes

moved some functions from WearableImpl to MessageHandler

deadYokai committed:
fix lint

deadYokai committed:
updated scary Bluetooth things

- and some DataItem changes

deadYokai committed:
Node & Assets db update

deadYokai committed:
reworked Bluetooth

deadYokai committed:
update code

deadYokai committed:
implement few unimplemented methods

teccheck committed:
Make use of packageName in ConnectionConfiguration

teccheck committed:
Add WearOS config UI

deadYokai commented:
@teccheck i can change my flash of ticwatch, but developer options is locked by function checkDebuggingDisallowed() in ClockworkSettings.apk and i dont know how to resign or/and replace it without bootloop

function content
```java
    private boolean checkDebuggingDisallowed() {
        RestrictedLockUtils.EnforcedAdmin enforcedAdmin = getEnforcedAdmin();
        if (enforcedAdmin != null) {
            RestrictedLockUtils.sendShowAdminSupportDetailsIntent(getActivity(), enforcedAdmin);
            return true;
        }
        if (!isUserBuild() || isSetupWizardCompleted()) {
            return isUserBuild() && isOemDebugRequired() && !isOemDebugEnabled();
        }
        return true;
   }
```

i tried changing init.rc params, but still cannot enable adb in setup

teccheck commented:
I'm relatively sure now that the disconnects have something to do with opening a channel:
<img width="1782" height="532" alt="grafik" src="https://github.com/user-attachments/assets/3bc5f351-7ae3-4f5b-85b7-3979e2c62833" />

The socket is always closed as soon as the channelRequest is sent. I don't think, there's anything wrong with the proto, but perhaps the values of the request are invalid somehow?

deadYokai commented:
what i managed to get from watch, maybe in other day will read logs in detail
```
03-01 15:15:47.734  2193  5560 D BluetoothSocket: Creating new Private BluetoothSocket of type: 1
03-01 15:15:47.734  2193  5560 D BluetoothSocket: acceptSocket: socket fd passed by stack fds:[java.io.FileDescriptor@634c532], mServiceName=WearableBt
03-01 15:15:47.735  2193  5560 I MigratingBTServer: setConnectable(false)
03-01 15:15:47.751  2079  2220 D AdapterProperties: Scan Mode:20
03-01 15:15:47.751  1426  1426 D WearBluetooth: *** BluetoothScanModeEnforcer-v01.3: Scan mode update Ignored by BluetoothScanModeEnforcer
03-01 15:15:47.753  2193  5560 I MigratingBTServer: Starting a connection with incoming address CE:02:46:3C:71:CC, addressMatchesNonMigratingConfig true, matched preexisting config: ConnectionConfiguration[ Name=server, Address=CE:02:46:3C:71:CC, Type=1, Role=2, Enabled=true, IsConnected=false, PeerNodeId=null, BtlePriority=true, NodeId=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, PackageName=null, ConnectionRetryStrategy=0, allowedConfigPackages=[], Migrating=false, DataItemSyncEnabled=true, NodeConnectionRestrictions=null, removeConnectionWhenBondRemovedByUser=true, maxSupportedRemoteAndroidSdkVersion=0]
03-01 15:15:47.754  2193  5560 I MigratingBTServer: This is a main config from addr CE:02:46:3C:71:CC, config: ConnectionConfiguration[ Name=server, Address=CE:02:46:3C:71:CC, Type=1, Role=2, Enabled=true, IsConnected=false, PeerNodeId=null, BtlePriority=true, NodeId=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, PackageName=null, ConnectionRetryStrategy=0, allowedConfigPackages=null, Migrating=false, DataItemSyncEnabled=true, NodeConnectionRestrictions=null, removeConnectionWhenBondRemovedByUser=true, maxSupportedRemoteAndroidSdkVersion=0]
03-01 15:15:47.759  2079  4452 D BtGatt.AdvertiseManager: stopAdvertisingSet() android.os.BinderProxy@e4cc9fd
03-01 15:15:47.759  2079  4452 I bt_stack: [INFO:le_advertising_manager.cc(70)] Unregister in shim layer
03-01 15:15:47.761  2193  5721 I WearBTServerConn: Connected to CE:02:46:3C:71:CC.
03-01 15:15:47.761  2193  5721 I Wear_Transport: handleConnection (enablePhoneSwitching: true) config: ConnectionConfiguration[ Name=server, Address=CE:02:46:3C:71:CC, Type=1, Role=2, Enabled=true, IsConnected=false, PeerNodeId=null, BtlePriority=true, NodeId=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, PackageName=null, ConnectionRetryStrategy=0, allowedConfigPackages=null, Migrating=false, DataItemSyncEnabled=true, NodeConnectionRestrictions=null, removeConnectionWhenBondRemovedByUser=true, maxSupportedRemoteAndroidSdkVersion=0]
03-01 15:15:47.762  2193  5721 I Wear_Transport: Connection continuing after step: auth/encryption
03-01 15:15:47.762  2193  5721 I Wear_Transport: Performing handshake with peer
03-01 15:15:47.817  2193  5721 I Wear_Transport: ClientNodeId = 71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5 connectNodeId = 71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5
03-01 15:15:47.819  2193  5721 I Wear_Transport: Connection continuing after step: handshake
03-01 15:15:47.819  2193  5721 I Wear_Transport: Setting 71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5 as nodeId connected to ConnectionConfiguration[ Name=server, Address=CE:02:46:3C:71:CC, Type=1, Role=2, Enabled=true, IsConnected=false, PeerNodeId=null, BtlePriority=true, NodeId=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, PackageName=null, ConnectionRetryStrategy=0, allowedConfigPackages=null, Migrating=false, DataItemSyncEnabled=true, NodeConnectionRestrictions=null, removeConnectionWhenBondRemovedByUser=true, maxSupportedRemoteAndroidSdkVersion=0]
03-01 15:15:47.822  2193  5721 I Wear_Transport: Connecting to config
03-01 15:15:47.844  2193  5721 V datatransport: doInitialDataSync: node=15f63c96, peer=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, sendingSyncStart=receivedSeqId=-1, syncTable={(15f63c96:208)}, version=2, restrictions=[]
03-01 15:15:47.845  2193  5721 I Wear_Transport: Connection continuing after step: connecting control plane to new node
03-01 15:15:47.849  2193  3019 I CloudSync: Disabling cloud sync.
03-01 15:15:47.852   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:47.884   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:47.885   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:47.942   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:47.973   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:47.974   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.057   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:48.543   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:48.543   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.555  2193  5723 V datatransport: respondToRemoteSyncStart: node=15f63c96, peer=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, remoteSyncStart=receivedSeqId=-1, syncTable={(71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5:33)}, version=2, restrictions=[]
03-01 15:15:48.586   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:48.586   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:48.586   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.627   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:48.652  2193  5724 V datatransport: handleSyncStart(SyncStartV2-71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5): node=15f63c96, peer=71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5, version=2, sent=78, synced={15f63c96=208, 71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5=33}, restrictions=[]
03-01 15:15:48.652  2193  5724 V datatransport: SyncStartV2-71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5 Completed sending data items in 87ms
03-01 15:15:48.669   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:48.669   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.734   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:48.748   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:48.748   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.879   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:48.910   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:48.910   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:48.987   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:49.008   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.008   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.074   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:49.093   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.093   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.159   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:49.183   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.183   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.244   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:49.253   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.256   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.258   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.258   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.308   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:49.357   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:49.358   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:49.452   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:50.284   910  2403 I vendor.qti.bluetooth@1.0-ibs_handler: DeviceSleep: TX Awake, Sending SLEEP_IND
03-01 15:15:50.284   910  2403 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK OFF
03-01 15:15:50.284   910  2403 D vendor.qti.bluetooth@1.0-uart_transport: ClockOperation: USERIAL_OP_CLK_OFF - after sys fs write-ret:4
03-01 15:15:50.435   910  2396 D vendor.qti.bluetooth@1.0-wake_lock: Release wakelock is released 
03-01 15:15:51.263  2079  2409 I bt_btm_pm: packages/modules/Bluetooth/system/stack/acl/btm_pm.cc:286 BTM_SetPowerMode: Setting power mode for peer:xx:xx:xx:xx:71:cc current_mode:immediate:active[0] new_mode:forced:sniff[2]
03-01 15:15:51.263  2079  2409 I bt_btm_pm: packages/modules/Bluetooth/system/stack/acl/btm_pm.cc:592 btm_pm_snd_md_req: Switching from immediate:active[0x00] to immediate:sniff[0x02]
03-01 15:15:51.264   910   910 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK ON
03-01 15:15:51.278   910   910 D vendor.qti.bluetooth@1.0-uart_transport: ClockOperation: USERIAL_OP_CLK_ON - after sys fs write-ret:4
03-01 15:15:51.278   910   910 D vendor.qti.bluetooth@1.0-wake_lock: Acquire wakelock is acquired 
03-01 15:15:51.278   910   910 I vendor.qti.bluetooth@1.0-ibs_handler: DeviceWakeUp: Writing IBS_WAKE_IND
03-01 15:15:51.279   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_ACK: 0xFC
03-01 15:15:51.279   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Signal wack_cond_
03-01 15:15:51.279   910   910 D vendor.qti.bluetooth@1.0-ibs_handler: DeviceWakeUp: Unblocked from waiting for FC, pthread_cond_timedwait ret = 0
03-01 15:15:51.281   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:15:51.281   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:15:51.282  2079  2409 I bt_btm_pm: packages/modules/Bluetooth/system/stack/acl/btm_pm.cc:706 btm_pm_proc_cmd_status: Clearing pending power mode link state:immediate:pending
03-01 15:15:51.282  2079  2409 E bluetooth: packages/modules/Bluetooth/system/bta/dm/bta_dm_pm.cc:1081 bta_dm_pm_btm_status: Received unknown power mode status event:5
03-01 15:15:51.298  2079  2386 I bluetooth: packages/modules/Bluetooth/system/gd/hci/hci_layer.cc:114 drop: Dropping event MAX_SLOTS_CHANGE
03-01 15:15:51.303  2079  2409 I bt_btm_pm: packages/modules/Bluetooth/system/stack/acl/btm_pm.cc:746 btm_pm_proc_mode_change: Power mode switched from immediate:pending[5] to immediate:sniff[2]
03-01 15:15:51.343   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
03-01 15:15:52.283   910  2403 I vendor.qti.bluetooth@1.0-ibs_handler: DeviceSleep: TX Awake, Sending SLEEP_IND
03-01 15:15:52.283   910  2403 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK OFF
03-01 15:15:52.284   910  2403 D vendor.qti.bluetooth@1.0-uart_transport: ClockOperation: USERIAL_OP_CLK_OFF - after sys fs write-ret:4
03-01 15:15:52.436   910  2396 D vendor.qti.bluetooth@1.0-wake_lock: Release wakelock is released 
03-01 15:15:56.003  2172  2172 D ActionBatteryChangeController: onReceive mBatteryReceiver
03-01 15:15:56.015  2172  2172 D ActionBatteryChangeController: onReceive mBatteryReceiver
03-01 15:16:02.577   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:16:02.577   910  2402 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK ON
03-01 15:16:02.578   910  2402 D vendor.qti.bluetooth@1.0-uart_transport: ClockOperation: USERIAL_OP_CLK_ON - after sys fs write-ret:4
03-01 15:16:02.578   910  2402 D vendor.qti.bluetooth@1.0-wake_lock: Acquire wakelock is acquired 
03-01 15:16:02.578   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:16:02.578   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
03-01 15:16:02.578   910  2402 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
03-01 15:16:02.592  2193  5721 I Wear_Transport: Threads terminated for node 71d2e1a0-2694-4b4d-a8f7-1f29f139c3c5
03-01 15:16:02.592  2193  5721 D BluetoothSocket: close() this: XX:XX:XX:3C:71:CC, channel: 6, mSocketIS: android.net.LocalSocketImpl$SocketInputStream@bdbc88a, mSocketOS: android.net.LocalSocketImpl$SocketOutputStream@86a9efb, mSocket: android.net.LocalSocket@2f85b18 impl:android.net.LocalSocketImpl@c104971 fd:java.io.FileDescriptor@634c532, mSocketState: CONNECTED, mServiceName=WearableBt
03-01 15:16:02.592  2193  5721 D BluetoothSocket: Closing mSocket: android.net.LocalSocket@2f85b18 impl:android.net.LocalSocketImpl@c104971 fd:java.io.FileDescriptor@634c532
03-01 15:16:02.593  2193  5721 D BluetoothSocket: close() this: XX:XX:XX:3C:71:CC, channel: 6, mSocketIS: android.net.LocalSocketImpl$SocketInputStream@bdbc88a, mSocketOS: android.net.LocalSocketImpl$SocketOutputStream@86a9efb, mSocket: null, mSocketState: CLOSED, mServiceName=WearableBt
```

teccheck committed:
Fix channel messages

deadYokai committed:
code update

deadYokai committed:
node migration update

deadYokai commented:
@teccheck check if okay, because i dont know git much

teccheck commented:
> discovered some proto differences for example, reversed from GMS

Nice!

deadYokai commented:
> I've made a clean version of the history in my branch. If you want, you can use that. If you want, you can even resign your last two commits (the `code update` commit is just your `code update + fixes by @teccheck` but without my edits)

hopefully i made it correct

teccheck commented:
Yes, almost. I think the last two commits are not needed. The revert commit is definitely not needed, as the commit it reverts does not exist in the history anymore.

I'm not sure what `Clean history does`. Did you make any changes after `node migration update`? 

If not, try `git reset HEAD~2 --hard`. This removes the last two commits from the history completely and resets your working directory to `node migration update`. This will require a force push

Also, I'm sorry for causing all this trouble. I hope this is ok for you…

deadYokai committed:
feat(wear): add missing messages to MessageHandler and MessageListener

teccheck committed:
feat(wear): add runtimeType to ConnectionConfiguration (field 20)

teccheck committed:
fix(wear): fix Connect message proto

teccheck committed:
feat(wear): add missing proto definitions

teccheck commented:
I've committed a few things to my branch. Please don't include them into your branch yet. I'll try to incorporate the good changes from 65a1e42bbf1d034929dd98aaccdd2c9f97098b1f

deadYokai committed:
feat(wear): add Wearable Reader and Writer

deadYokai committed:
feat(wear): add NetworkConnectionManager

deadYokai committed:
rough ble implementation

deadYokai commented:
> I just noticed, that I made a mistake with the ChannelControl Type in proto. They should be:
> 
> ```java
> CHANNEL_CONTROL_OPEN = 1;
> CHANNEL_CONTROL_OPEN_ACK = 2;
> CHANNEL_CONTROL_CLOSE = 3;
> ```
> 
> This was correct in the constants in `ChannelManager`, but with your last two commits that changed.

<img width="540" height="304" alt="image" src="https://github.com/user-attachments/assets/ffb09aab-ee4e-4693-afff-436ad9b880b6" />

deadYokai committed:
fix Type enum in wearable.proto

deadYokai committed:
account matching stub

deadYokai committed:
minor changes

deadYokai committed:
added TransportConnectionHandler class

deadYokai committed:
integrate TransportConnectionHandler class

deadYokai committed:
added DataTransport class

deadYokai committed:
integration of DataTransport class

deadYokai committed:
added of AssetManager class

deadYokai committed:
integration of AssetManager class

deadYokai committed:
a2a encryption stub draft

deadYokai committed:
move Connect handshake to separate class

deadYokai committed:
Connect handshake rework

deadYokai committed:
updated NodeMigration classes

deadYokai committed:
updated NodeMigration

deadYokai committed:
added missing peerNodeId to MessageHandler

deadYokai committed:
update and fixes

deadYokai committed:
Merge remote-tracking branch 'upstream/master' into wearable_tos

deadYokai committed:
updated BleConnectionManager

deadYokai committed:
Ble wiring

deadYokai committed:
updated NodeDatabaseHelper

deadYokai committed:
bug fixes

deadYokai committed:
bug fixes & added peernodeid to db

deadYokai committed:
fixes nodeid and peernodeid assigments

deadYokai committed:
set watch nodeid to uuid instead of 8 hex

deadYokai committed:
update

deadYokai committed:
enable connection in putConfig

deadYokai committed:
deduplicate config if node name is address

deadYokai committed:
bug fixes

deadYokai commented:
still can't get past the "Continue on phone" step :(

deadYokai committed:
intent updates

deadYokai committed:
fix fd leak, cached streams per fd

deadYokai committed:
send close request to channels before bt drop

deadYokai committed:
[ui] added drop connectionconfiguration table button
and fixed typo `prefcat_wear_connections` -> `pref_cat_wear_connections`
useful for broken entries, that cannot be deleted using standard methods

deadYokai committed:
added forgotten peernodeid to onCreate in ConfigurationDatabaseHelper

deadYokai committed:
fixes

deadYokai committed:
update

deadYokai commented:
my logs from companion app (hooked via frida)
```
[LogUtil][D] MobvoiApp: onEnterForeground
[LogUtil][D] TicCareDeviceManager: onEnterForeground
[LogUtil][E] SetupNavigator: onResume null
[LogUtil][I] TicCareDeviceManager:  BroadcastReceiver  action = action.app_foregound
[LogUtil][I] TicCareDeviceManager:  getDeviceFromServer 
[LogUtil][D] TOSFragment: Callback in termsOfServiceStatusObserver TermsAccepted(cloudSyncAccepted=false, loggingAccepted=false, locationAccepted=false)
[LogUtil][D] SetupNavigator: SDK setup step changed
[LogUtil][D] SetupNavigator: navigate() WatchDiscoverySetupStep(available=true, status=NotStarted)
[LogUtil][D] SetupNavigator: WatchDiscoveryFragment
[LogUtil][D] SetupNavigator: bind
[LogUtil][D] SetupNavigator: shareWatchView
[LogUtil][D] WatchDiscoveryViewModel: startWatchDiscovery()
[LogUtil][D] WatchDiscovery: WatchDiscoverySetupStep.Status.Scanning []
[LogUtil][D] WatchDiscoveryAdapter: addDevices 1
[LogUtil][D] WatchDiscovery: WatchDiscoverySetupStep.Status.Scanning [DiscoveredDevice(name=TicWatch Pro 5 0872, address=30:95:E3:E8:D8:86, watchPreviousPairingStatus=c)]
[LogUtil][D] WatchDiscoveryAdapter: addDevices 3
[LogUtil][D] WatchDiscoveryAdapter: DeviceInfo onClickListener
[LogUtil][D] WatchDiscovery: onDeviceSelected device DiscoveredDevice(name=TicWatch Pro 5 0872, address=30:95:E3:E8:D8:86, watchPreviousPairingStatus=c)
[LogUtil][D] WatchDiscoveryViewModel: selectDevice
[LogUtil][D] WatchDiscovery: WatchDiscoverySetupStep.Status.NotStarted
[LogUtil][D] SetupNavigator: SDK setup step changed
[LogUtil][D] SetupNavigator: navigate() WatchPairingSetupStep(available=true, status=NotStarted)
[LogUtil][D] SetupNavigator: WatchPairingFragment
[LogUtil][D] SetupNavigator: bind
[LogUtil][D] SetupNavigator: shareWatchView
[LogUtil][D] WatchPairingFragment: WatchPairingFragment.Status.InProgress
[LogUtil][E] SetupNavigator: onPause
[LogUtil][D] MobvoiApp: onEnterBackground
[LogUtil][D] TicCareDeviceManager: onEnterBackground
[LogUtil][D] MobvoiApp: onEnterForeground
[LogUtil][D] TicCareDeviceManager: onEnterForeground
[LogUtil][E] SetupNavigator: onResume null
[LogUtil][I] TicCareDeviceManager:  BroadcastReceiver  action = action.app_foregound
[LogUtil][I] TicCareDeviceManager:  getDeviceFromServer 
[LogUtil][D] WatchPairingFragment: WatchPairingFragment.Status.Success
[LogUtil][I] WatchPairingViewModel: onPairingSuccess()
[LogUtil][D] SetupNavigator: SDK setup step changed
[LogUtil][D] SetupNavigator: navigate() CompanionDeviceManagerSetupStep(available=true, status=NotStarted)
[LogUtil][D] SetupNavigator: CompanionDeviceManagerFragment
[LogUtil][D] SetupNavigator: bind
[LogUtil][D] SetupNavigator: shareWatchView
[LogUtil][D] CdmFragment: startAssociation
[LogUtil][I] CdmViewModel: startAssociation
[LogUtil][D] CdmFragment: Status.Success
[LogUtil][I] CdmViewModel: onSuccess
[LogUtil][D] SetupNavigator: SDK setup step changed
[LogUtil][D] SetupNavigator: navigate() WatchConnectionSetupStep(available=true, status=InProgress(progress=0))
[LogUtil][D] SetupNavigator: WatchConnectionFragment
[LogUtil][D] SetupNavigator: bind
[LogUtil][D] WatchConnectionFragment: Status.InProgress 0
[LogUtil][D] WatchConnectionFragment: Status.InProgress 12
[LogUtil][D] WatchConnectionFragment: Status.InProgress 25
[LogUtil][D] GmsWLService: onMessageReceived: /health/cmd_new_wear_push_to_phone, node: 514cbd70
[LogUtil][D] MessageProxyClient: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] WearMessageReceiver: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] health.bt.listener: onMessageReceived /health/cmd_new_wear_push_to_phone
[LogUtil][D] health.HealthDataSync:  saveWearData {"current_step":0,"goal":10000}
[LogUtil][D] GmsWearableClient: sendMessage, mConnectState 3 mConnectedNodesInitialized true
[LogUtil][D] GmsWearableClient: sendMessage, nodeId=[*], path=[/health/cmd_new_wear_push_to_phone], nodes=[NodeInfo[nodeType=1, nodeId=b17d18c7, nodeName=b17d18c7, nearby=false], NodeInfo[nodeType=1, nodeId=15f63c96, nodeName=15f63c96, nearby=false], NodeInfo[nodeType=1, nodeId=a9228cca, nodeName=a9228cca, nearby=false], NodeInfo[nodeType=1, nodeId=e353b7bc, nodeName=e353b7bc, nearby=false], NodeInfo[nodeType=1, nodeId=64382908, nodeName=64382908, nearby=false], NodeInfo[nodeType=1, nodeId=9edb3636, nodeName=9edb3636, nearby=false], NodeInfo[nodeType=1, nodeId=b4f2b948, nodeName=b4f2b948, nearby=false], NodeInfo[nodeType=1, nodeId=89d864df, nodeName=89d864df, nearby=false], NodeInfo[nodeType=1, nodeId=88a81867, nodeName=88a81867, nearby=false], NodeInfo[nodeType=1, nodeId=75f326ff, nodeName=75f326ff, nearby=false], NodeInfo[nodeType=1, nodeId=84f95e13, nodeName=84f95e13, nearby=false], NodeInfo[nodeType=1, nodeId=b41fa1a7, nodeName=b41fa1a7, nearby=false], NodeInfo[nodeType=1, nodeId=45859bcd, nodeName=45859bcd, nearby=false], NodeInfo[nodeType=1, nodeId=5e0f73d, nodeName=5e0f73d, nearby=false], NodeInfo[nodeType=1, nodeId=ae19a8fc, nodeName=ae19a8fc, nearby=false], NodeInfo[nodeType=1, nodeId=3605162e, nodeName=3605162e, nearby=false], NodeInfo[nodeType=1, nodeId=f8986be8, nodeName=f8986be8, nearby=false], NodeInfo[nodeType=1, nodeId=375e0681, nodeName=375e0681, nearby=false], NodeInfo[nodeType=1, nodeId=8c3eeec0, nodeName=8c3eeec0, nearby=false], NodeInfo[nodeType=1, nodeId=5cb3ea22, nodeName=5cb3ea22, nearby=false], NodeInfo[nodeType=1, nodeId=58e59c53, nodeName=58e59c53, nearby=false], NodeInfo[nodeType=1, nodeId=82f80424, nodeName=82f80424, nearby=false], NodeInfo[nodeType=1, nodeId=acb95ac1, nodeName=TicWatch Pro 5 0872, nearby=false], NodeInfo[nodeType=1, nodeId=71083995, nodeName=71083995, nearby=false], NodeInfo[nodeType=1, nodeId=3533ff6b, nodeName=3533ff6b, nearby=false], NodeInfo[nodeType=1, nodeId=25089bbf, nodeName=25089bbf, nearby=false], NodeInfo[nodeType=1, nodeId=33c8ef3b, nodeName=33c8ef3b, nearby=false], NodeInfo[nodeType=1, nodeId=8ccad6, nodeName=8ccad6, nearby=false], NodeInfo[nodeType=1, nodeId=63b9a046, nodeName=63b9a046, nearby=false], NodeInfo[nodeType=1, nodeId=c91c5922, nodeName=c91c5922, nearby=false], NodeInfo[nodeType=1, nodeId=6e47438e, nodeName=6e47438e, nearby=false], NodeInfo[nodeType=1, nodeId=cad685e, nodeName=cad685e, nearby=false], NodeInfo[nodeType=1, nodeId=648da173, nodeName=648da173, nearby=false], NodeInfo[nodeType=1, nodeId=cbb758b2, nodeName=cbb758b2, nearby=false], NodeInfo[nodeType=1, nodeId=fc02e5bb, nodeName=fc02e5bb, nearby=false], NodeInfo[nodeType=1, nodeId=698965e6, nodeName=698965e6, nearby=false], NodeInfo[nodeType=1, nodeId=d0143381, nodeName=d0143381, nearby=false], NodeInfo[nodeType=1, nodeId=ed5164ba, nodeName=ed5164ba, nearby=false], NodeInfo[nodeType=1, nodeId=fd1b95d8, nodeName=fd1b95d8, nearby=false], NodeInfo[nodeType=1, nodeId=25346cd2, nodeName=25346cd2, nearby=false], NodeInfo[nodeType=1, nodeId=d70b905a, nodeName=d70b905a, nearby=false], NodeInfo[nodeType=1, nodeId=8df461d7, nodeName=8df461d7, nearby=false], NodeInfo[nodeType=1, nodeId=d3e36edc, nodeName=d3e36edc, nearby=false], NodeInfo[nodeType=1, nodeId=7803d38a, nodeName=7803d38a, nearby=false]]
[LogUtil][D] health.HealthDataSync: sync steps 0
[LogUtil][E] WechatSportUtil: wwid is null, deviceId is , step is 0
[LogUtil][D] GmsWLService: onMessageReceived: /companion/sync_wear_info, node: 514cbd70
[LogUtil][D] MessageProxyClient: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/companion/sync_wear_info]
[LogUtil][D] WearMessageReceiver: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/companion/sync_wear_info]
[LogUtil][D] WearMessageReceiver: updateWearInfo: nodeId 514cbd70, data {"btAddress":"30:95:E3:E8:D8:86","btName":"TicWatch Pro 5","certModel":"WH12088","hasNfcFeature":true,"isMfiSupported":true,"region":"global","skuColor":"","skuTheme":"","sn":"C101X36150872","ticwatchChannel":"","wearCapability":"fitness_hr_warning|health_direct_mms|health_auto_motion|health_support_delete|health_bind_wechat|fitness_rowing_machine|health_cloud_sync|health_protocol_v2|fitness_protocol_v2|privacy_cloud_sync|fitness_indoor_cycling|account_fork|heart_background_tracking|fitness_compressed_data|fitness_cloud_sync|watch_sensor_heart_rate|watch_sensor_act_recognition|tic_sos|health_hr_warning|fitness_auto_start","wearDeviceId":"[redacted]","wearType":"TicWatch Pro 5","wearVersionNumber":33}
[LogUtil][D] WearPairingPool: updateWearInfo: nodeId 514cbd70, data {"btAddress":"30:95:E3:E8:D8:86","btName":"TicWatch Pro 5","certModel":"WH12088","hasNfcFeature":true,"isMfiSupported":true,"region":"global","skuColor":"","skuTheme":"","sn":"C101X36150872","ticwatchChannel":"","wearCapability":"fitness_hr_warning|health_direct_mms|health_auto_motion|health_support_delete|health_bind_wechat|fitness_rowing_machine|health_cloud_sync|health_protocol_v2|fitness_protocol_v2|privacy_cloud_sync|fitness_indoor_cycling|account_fork|heart_background_tracking|fitness_compressed_data|fitness_cloud_sync|watch_sensor_heart_rate|watch_sensor_act_recognition|tic_sos|health_hr_warning|fitness_auto_start","wearDeviceId":"[redacted]","wearType":"TicWatch Pro 5","wearVersionNumber":33}
[LogUtil][D] WearPairingPool: Update deviceId of 514cbd70 to [redacted]
[LogUtil][D] GmsWLService: onMessageReceived: /health/cmd_new_wear_push_to_phone, node: 514cbd70
[LogUtil][D] WearInfoHelper: Remove nodes []
[LogUtil][D] WearInfoHelper: Add nodes info for 514cbd70
[LogUtil][D] MessageProxyClient: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] WearMessageReceiver: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] health.bt.listener: onMessageReceived /health/cmd_new_wear_push_to_phone
[LogUtil][D] health.HealthDataSync:  saveWearData {"current_step":0,"goal":10000}
[LogUtil][D] GmsWearableClient: sendMessage, mConnectState 3 mConnectedNodesInitialized true
[LogUtil][D] GmsWearableClient: sendMessage, nodeId=[*], path=[/health/cmd_new_wear_push_to_phone], nodes=[NodeInfo[nodeType=1, nodeId=b17d18c7, nodeName=b17d18c7, nearby=false], NodeInfo[nodeType=1, nodeId=15f63c96, nodeName=15f63c96, nearby=false], NodeInfo[nodeType=1, nodeId=a9228cca, nodeName=a9228cca, nearby=false], NodeInfo[nodeType=1, nodeId=e353b7bc, nodeName=e353b7bc, nearby=false], NodeInfo[nodeType=1, nodeId=64382908, nodeName=64382908, nearby=false], NodeInfo[nodeType=1, nodeId=9edb3636, nodeName=9edb3636, nearby=false], NodeInfo[nodeType=1, nodeId=b4f2b948, nodeName=b4f2b948, nearby=false], NodeInfo[nodeType=1, nodeId=89d864df, nodeName=89d864df, nearby=false], NodeInfo[nodeType=1, nodeId=88a81867, nodeName=88a81867, nearby=false], NodeInfo[nodeType=1, nodeId=75f326ff, nodeName=75f326ff, nearby=false], NodeInfo[nodeType=1, nodeId=84f95e13, nodeName=84f95e13, nearby=false], NodeInfo[nodeType=1, nodeId=b41fa1a7, nodeName=b41fa1a7, nearby=false], NodeInfo[nodeType=1, nodeId=45859bcd, nodeName=45859bcd, nearby=false], NodeInfo[nodeType=1, nodeId=5e0f73d, nodeName=5e0f73d, nearby=false], NodeInfo[nodeType=1, nodeId=ae19a8fc, nodeName=ae19a8fc, nearby=false], NodeInfo[nodeType=1, nodeId=3605162e, nodeName=3605162e, nearby=false], NodeInfo[nodeType=1, nodeId=f8986be8, nodeName=f8986be8, nearby=false], NodeInfo[nodeType=1, nodeId=375e0681, nodeName=375e0681, nearby=false], NodeInfo[nodeType=1, nodeId=8c3eeec0, nodeName=8c3eeec0, nearby=false], NodeInfo[nodeType=1, nodeId=5cb3ea22, nodeName=5cb3ea22, nearby=false], NodeInfo[nodeType=1, nodeId=58e59c53, nodeName=58e59c53, nearby=false], NodeInfo[nodeType=1, nodeId=82f80424, nodeName=82f80424, nearby=false], NodeInfo[nodeType=1, nodeId=acb95ac1, nodeName=TicWatch Pro 5 0872, nearby=false], NodeInfo[nodeType=1, nodeId=71083995, nodeName=71083995, nearby=false], NodeInfo[nodeType=1, nodeId=3533ff6b, nodeName=3533ff6b, nearby=false], NodeInfo[nodeType=1, nodeId=25089bbf, nodeName=25089bbf, nearby=false], NodeInfo[nodeType=1, nodeId=33c8ef3b, nodeName=33c8ef3b, nearby=false], NodeInfo[nodeType=1, nodeId=8ccad6, nodeName=8ccad6, nearby=false], NodeInfo[nodeType=1, nodeId=63b9a046, nodeName=63b9a046, nearby=false], NodeInfo[nodeType=1, nodeId=c91c5922, nodeName=c91c5922, nearby=false], NodeInfo[nodeType=1, nodeId=6e47438e, nodeName=6e47438e, nearby=false], NodeInfo[nodeType=1, nodeId=cad685e, nodeName=cad685e, nearby=false], NodeInfo[nodeType=1, nodeId=648da173, nodeName=648da173, nearby=false], NodeInfo[nodeType=1, nodeId=cbb758b2, nodeName=cbb758b2, nearby=false], NodeInfo[nodeType=1, nodeId=fc02e5bb, nodeName=fc02e5bb, nearby=false], NodeInfo[nodeType=1, nodeId=698965e6, nodeName=698965e6, nearby=false], NodeInfo[nodeType=1, nodeId=d0143381, nodeName=d0143381, nearby=false], NodeInfo[nodeType=1, nodeId=ed5164ba, nodeName=ed5164ba, nearby=false], NodeInfo[nodeType=1, nodeId=fd1b95d8, nodeName=fd1b95d8, nearby=false], NodeInfo[nodeType=1, nodeId=25346cd2, nodeName=25346cd2, nearby=false], NodeInfo[nodeType=1, nodeId=d70b905a, nodeName=d70b905a, nearby=false], NodeInfo[nodeType=1, nodeId=8df461d7, nodeName=8df461d7, nearby=false], NodeInfo[nodeType=1, nodeId=d3e36edc, nodeName=d3e36edc, nearby=false], NodeInfo[nodeType=1, nodeId=7803d38a, nodeName=7803d38a, nearby=false]]
[LogUtil][D] health.HealthDataSync: sync steps 0
[LogUtil][E] WechatSportUtil: wwid is null, deviceId is , step is 0
[LogUtil][D] GmsWLService: onMessageReceived: /companion/sync_wear_info, node: 514cbd70
[LogUtil][D] MessageProxyClient: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/companion/sync_wear_info]
[LogUtil][D] WearMessageReceiver: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/companion/sync_wear_info]
[LogUtil][D] WearMessageReceiver: updateWearInfo: nodeId 514cbd70, data {"btAddress":"30:95:E3:E8:D8:86","btName":"TicWatch Pro 5","certModel":"WH12088","hasNfcFeature":true,"isMfiSupported":true,"region":"global","skuColor":"","skuTheme":"","sn":"C101X36150872","ticwatchChannel":"","wearCapability":"fitness_hr_warning|health_direct_mms|health_auto_motion|health_support_delete|health_bind_wechat|fitness_rowing_machine|health_cloud_sync|health_protocol_v2|fitness_protocol_v2|privacy_cloud_sync|fitness_indoor_cycling|account_fork|heart_background_tracking|fitness_compressed_data|fitness_cloud_sync|watch_sensor_heart_rate|watch_sensor_act_recognition|tic_sos|health_hr_warning|fitness_auto_start","wearDeviceId":"[redacted]","wearType":"TicWatch Pro 5","wearVersionNumber":33}
[LogUtil][D] WearPairingPool: updateWearInfo: nodeId 514cbd70, data {"btAddress":"30:95:E3:E8:D8:86","btName":"TicWatch Pro 5","certModel":"WH12088","hasNfcFeature":true,"isMfiSupported":true,"region":"global","skuColor":"","skuTheme":"","sn":"C101X36150872","ticwatchChannel":"","wearCapability":"fitness_hr_warning|health_direct_mms|health_auto_motion|health_support_delete|health_bind_wechat|fitness_rowing_machine|health_cloud_sync|health_protocol_v2|fitness_protocol_v2|privacy_cloud_sync|fitness_indoor_cycling|account_fork|heart_background_tracking|fitness_compressed_data|fitness_cloud_sync|watch_sensor_heart_rate|watch_sensor_act_recognition|tic_sos|health_hr_warning|fitness_auto_start","wearDeviceId":"[redacted]","wearType":"TicWatch Pro 5","wearVersionNumber":33}
[LogUtil][D] WearPairingPool: Update deviceId of 514cbd70 to [redacted]
[LogUtil][D] GmsWLService: onMessageReceived: /health/cmd_new_wear_push_to_phone, node: 514cbd70
[LogUtil][D] WearInfoHelper: Remove nodes []
[LogUtil][D] WearInfoHelper: Add nodes info for 514cbd70
[LogUtil][D] MessageProxyClient: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] WearMessageReceiver: onMessageReceived: MessageInfo[nodeId=514cbd70, path=/health/cmd_new_wear_push_to_phone]
[LogUtil][D] health.bt.listener: onMessageReceived /health/cmd_new_wear_push_to_phone
[LogUtil][D] health.HealthDataSync:  saveWearData {"current_step":0,"goal":10000}
[LogUtil][D] GmsWearableClient: sendMessage, mConnectState 3 mConnectedNodesInitialized true
[LogUtil][D] GmsWearableClient: sendMessage, nodeId=[*], path=[/health/cmd_new_wear_push_to_phone], nodes=[NodeInfo[nodeType=1, nodeId=b17d18c7, nodeName=b17d18c7, nearby=false], NodeInfo[nodeType=1, nodeId=15f63c96, nodeName=15f63c96, nearby=false], NodeInfo[nodeType=1, nodeId=a9228cca, nodeName=a9228cca, nearby=false], NodeInfo[nodeType=1, nodeId=e353b7bc, nodeName=e353b7bc, nearby=false], NodeInfo[nodeType=1, nodeId=64382908, nodeName=64382908, nearby=false], NodeInfo[nodeType=1, nodeId=9edb3636, nodeName=9edb3636, nearby=false], NodeInfo[nodeType=1, nodeId=b4f2b948, nodeName=b4f2b948, nearby=false], NodeInfo[nodeType=1, nodeId=89d864df, nodeName=89d864df, nearby=false], NodeInfo[nodeType=1, nodeId=88a81867, nodeName=88a81867, nearby=false], NodeInfo[nodeType=1, nodeId=75f326ff, nodeName=75f326ff, nearby=false], NodeInfo[nodeType=1, nodeId=84f95e13, nodeName=84f95e13, nearby=false], NodeInfo[nodeType=1, nodeId=b41fa1a7, nodeName=b41fa1a7, nearby=false], NodeInfo[nodeType=1, nodeId=45859bcd, nodeName=45859bcd, nearby=false], NodeInfo[nodeType=1, nodeId=5e0f73d, nodeName=5e0f73d, nearby=false], NodeInfo[nodeType=1, nodeId=ae19a8fc, nodeName=ae19a8fc, nearby=false], NodeInfo[nodeType=1, nodeId=3605162e, nodeName=3605162e, nearby=false], NodeInfo[nodeType=1, nodeId=f8986be8, nodeName=f8986be8, nearby=false], NodeInfo[nodeType=1, nodeId=375e0681, nodeName=375e0681, nearby=false], NodeInfo[nodeType=1, nodeId=8c3eeec0, nodeName=8c3eeec0, nearby=false], NodeInfo[nodeType=1, nodeId=5cb3ea22, nodeName=5cb3ea22, nearby=false], NodeInfo[nodeType=1, nodeId=58e59c53, nodeName=58e59c53, nearby=false], NodeInfo[nodeType=1, nodeId=82f80424, nodeName=82f80424, nearby=false], NodeInfo[nodeType=1, nodeId=acb95ac1, nodeName=TicWatch Pro 5 0872, nearby=false], NodeInfo[nodeType=1, nodeId=71083995, nodeName=71083995, nearby=false], NodeInfo[nodeType=1, nodeId=3533ff6b, nodeName=3533ff6b, nearby=false], NodeInfo[nodeType=1, nodeId=25089bbf, nodeName=25089bbf, nearby=false], NodeInfo[nodeType=1, nodeId=33c8ef3b, nodeName=33c8ef3b, nearby=false], NodeInfo[nodeType=1, nodeId=8ccad6, nodeName=8ccad6, nearby=false], NodeInfo[nodeType=1, nodeId=63b9a046, nodeName=63b9a046, nearby=false], NodeInfo[nodeType=1, nodeId=c91c5922, nodeName=c91c5922, nearby=false], NodeInfo[nodeType=1, nodeId=6e47438e, nodeName=6e47438e, nearby=false], NodeInfo[nodeType=1, nodeId=cad685e, nodeName=cad685e, nearby=false], NodeInfo[nodeType=1, nodeId=648da173, nodeName=648da173, nearby=false], NodeInfo[nodeType=1, nodeId=cbb758b2, nodeName=cbb758b2, nearby=false], NodeInfo[nodeType=1, nodeId=fc02e5bb, nodeName=fc02e5bb, nearby=false], NodeInfo[nodeType=1, nodeId=698965e6, nodeName=698965e6, nearby=false], NodeInfo[nodeType=1, nodeId=d0143381, nodeName=d0143381, nearby=false], NodeInfo[nodeType=1, nodeId=ed5164ba, nodeName=ed5164ba, nearby=false], NodeInfo[nodeType=1, nodeId=fd1b95d8, nodeName=fd1b95d8, nearby=false], NodeInfo[nodeType=1, nodeId=25346cd2, nodeName=25346cd2, nearby=false], NodeInfo[nodeType=1, nodeId=d70b905a, nodeName=d70b905a, nearby=false], NodeInfo[nodeType=1, nodeId=8df461d7, nodeName=8df461d7, nearby=false], NodeInfo[nodeType=1, nodeId=d3e36edc, nodeName=d3e36edc, nearby=false], NodeInfo[nodeType=1, nodeId=7803d38a, nodeName=7803d38a, nearby=false]]
[LogUtil][D] health.HealthDataSync: sync steps 0
[LogUtil][E] WechatSportUtil: wwid is null, deviceId is , step is 0
```

deadYokai committed:
update

deadYokai committed:
update

deadYokai committed:
update

deadYokai commented:
now stuck at `InProgress → SET_TETHER_CONFIG(62%)`

for success needed:
  62%: "SET_TETHER_CONFIG",
  75%: "SYNC_TIME_ON_WATCH",
  87%: "SYNC_LANGUAGE_ON_WATCH",
  100%: "REFRESH_WATCH_API",

deadYokai committed:
update

deadYokai commented:
now crash after "Connect account google"

Edit: @teccheck can you check on your side? because i only use Mobvoi Health for testing, not WearOS app

deadYokai committed:
update

many changes in WearableImpl mostly caused by "Reformat a file" in AndroidStudio

deadYokai committed:
update

deadYokai commented:
after https://github.com/microg/GmsCore/pull/3204/commits/5e6d977ee17fd4dc6df230619311fba778b6f7b3 
successfully synced language

Edit: maybe in https://github.com/microg/GmsCore/pull/3204/commits/fef3df7355caee92ebdf2d7e7642a0c9905cbde4 or https://github.com/microg/GmsCore/pull/3204/commits/77e48cb502da883087ba00d88f556bf4ad536ea7 because i didn't notice it at first

teccheck commented:
For me, it still hangs at Connecting to your watch. The logs end at `GmsWearSvcImpl: onTransact [unknown]: 72, android.os.Parcel@7ee7f1c, 18`. Seems like the call with id 72 is missing from `IWearableService.aidl`. Will see, what I can find out

deadYokai commented:
> For me, it still hangs at Connecting to your watch. The logs end at `GmsWearSvcImpl: onTransact [unknown]: 72, android.os.Parcel@7ee7f1c, 18`. Seems like the call with id 72 is missing from `IWearableService.aidl`. Will see, what I can find out

hmm, strange, because is present in IWearableService, maybe not implemented or without stub?
```java
    void getRelatedConfigs(IWearableCallbacks callbacks) = 72;
    void updateConfig(IWearableCallbacks iWearableCallbacks, in ConnectionConfiguration config) = 73;
```

Edit: maybe IWearableCallbacks?

Edit 2: in original gms IWearableCallbacks only 50 entries

Edit 3: i miscalculated, missing entry 71 (maybe)  -- in my jadx is named as updateConnectionStrategy

deadYokai committed:
updateConnectionStrategy.aidl stub

deadYokai committed:
update

deadYokai commented:
later i properly implement updateConnectionStrategy, if its our missing entry

teccheck commented:
> hmm, strange, because is present in IWearableService, maybe not implemented or without stub?

Not for me, I'm on commit 5e6d977ee17fd4dc6df230619311fba778b6f7b3. Are you sure, you've committed everything?

EDIT: I'm stupid, nevermind

teccheck commented:
> Edit 3: i miscalculated, missing entry 71 (maybe) -- in my jadx is named as updateConnectionStrategy

That's likely it

deadYokai committed:
implementation of updateConnectionStrategy

deadYokai committed:
implementation of backup/restore parcelables

deadYokai committed:
implementation of privacyRecordOptinRequest

deadYokai committed:
implementation of dataSynchronizationProgressTracking

and syncWifiCredentialWithSsid, syncWifiCredentialForNode stubs

deadYokai committed:
update

and added IosMultiAppAuth to proto

deadYokai committed:
update & bug fixes

deadYokai committed:
update

and added IosMultiAppAuth to proto

deadYokai commented:
almost

<img height="720"  alt="Screenshot_20260529-035707_Mobvoi_Health" src="https://github.com/user-attachments/assets/8f1e86f5-26a8-40ea-94aa-50fb4f2e23c6" />

deadYokai committed:
DataTransport update

deadYokai committed:
DataTransport exclude ghost nodes

deadYokai committed:
remove custom heartbeat that can cause connection issues

deadYokai committed:
update
